comp/syntheticstestscheduler/impl/worker.go

```diff
@@ -0,0 +1,342 @@
+type WorkerResult struct {
+	tracerouteResult payload.NetworkPath
+	tracerouteError  error
+	testCfg          SyntheticsTestCtx
+
+	triggeredAt   time.Time
+	startedAt     time.Time
+	finishedAt    time.Time
+	duration      time.Duration
+	tracerouteCfg config.Config
+}
+
+// toNetpathConfig converts a SyntheticsTestConfig into a system-probe Config.
+func toNetpathConfig(c common.SyntheticsTestConfig) (config.Config, error) {
+	var cfg config.Config
+
+	switch common.SubType(c.Subtype) {
+	case common.SubTypeUDP:
+		req, ok := c.Config.Request.(common.UDPConfigRequest)
+		if !ok {
+			return config.Config{}, fmt.Errorf("invalid UDP request type")
+		}
+		cfg.Protocol = payload.ProtocolUDP
+		cfg.DestHostname = req.Host
+		if req.Port != nil {
+			cfg.DestPort = uint16(*req.Port)
+		}
+		if req.MaxTTL != nil {
+			cfg.MaxTTL = uint8(*req.MaxTTL)
+		}
+		if req.Timeout != nil {
+			cfg.Timeout = time.Duration(*req.Timeout) * time.Second
+		}
+		if req.SourceService != nil {
+			cfg.SourceService = *req.SourceService
+		}
+		if req.DestinationService != nil {
+			cfg.DestinationService = *req.DestinationService
+		}
+
+	case common.SubTypeTCP:
+		req, ok := c.Config.Request.(common.TCPConfigRequest)
+		if !ok {
+			return config.Config{}, fmt.Errorf("invalid TCP request type")
+		}
+		cfg.Protocol = payload.ProtocolTCP
+		cfg.DestHostname = req.Host
+		if req.Port != nil {
+			cfg.DestPort = uint16(*req.Port)
+		}
+		if req.MaxTTL != nil {
+			cfg.MaxTTL = uint8(*req.MaxTTL)
+		}
+		if req.Timeout != nil {
+			cfg.Timeout = time.Duration(*req.Timeout) * time.Second
+		}
+		cfg.TCPMethod = payload.TCPMethod(req.TCPMethod)
+		if req.SourceService != nil {
+			cfg.SourceService = *req.SourceService
+		}
+		if req.DestinationService != nil {
+			cfg.DestinationService = *req.DestinationService
+		}
+
+	case common.SubTypeICMP:
+		req, ok := c.Config.Request.(common.ICMPConfigRequest)
+		if !ok {
+			return config.Config{}, fmt.Errorf("invalid ICMP request type")
+		}
+		cfg.Protocol = payload.ProtocolICMP
+		cfg.DestHostname = req.Host
+		if req.MaxTTL != nil {
+			cfg.MaxTTL = uint8(*req.MaxTTL)
+		}
+		if req.Timeout != nil {
+			cfg.Timeout = time.Duration(*req.Timeout) * time.Second
+		}
+		if req.SourceService != nil {
+			cfg.SourceService = *req.SourceService
+		}
+		if req.DestinationService != nil {
+			cfg.DestinationService = *req.DestinationService
+		}
+
+	default:
+		return config.Config{}, fmt.Errorf("unsupported subtype: %s", c.Subtype)
+	}
+
+	return cfg, nil
+}
+
+// SyntheticsTestCtx is the unit of work consumed by workers.
+type SyntheticsTestCtx struct {
+	nextRun time.Time
+	cfg     common.SyntheticsTestConfig
+}
+
+// updateTestState updates lastRun and nextRun for a running test.
+func (s *SyntheticsTestScheduler) updateTestState(flushTime time.Time, rt *runningTestState) {
+	rt.lastRun = flushTime
+	rt.nextRun = flushTime.Add(time.Duration(rt.cfg.Interval) * time.Second)
```

Comment:
`flushTime` might vary when the agent is overloaded for example, shall we rely only on the nextRun/lastRun info?

For example, something like

```suggestion
	rt.nextRun = rt.nextRun.Add(time.Duration(rt.cfg.Interval) * time.Second)
	rt.lastRun = rt.nextRun
```

This way, there is less risk of "drifting" if current flush time is not precise.